src/main/java/apoc/merge/Merge.java

```diff
@@ -25,26 +25,49 @@
+        params.put("props", props);
         String identPropsString = buildIdentPropsString(identProps);
 
         final String cypher = "MERGE (n:" + labels + "{" + identPropsString + "}) ON CREATE SET n += $props RETURN n";
         Node node = Iterators.single(db.execute(cypher, params ).columnAs("n"));
         return Stream.of(new NodeResult(node));
     }
 
-    private String wrapInBacktics(String s) {
-        return "`" + s + "`";
+    @Procedure(value="apoc.merge.node.eager", mode = Mode.WRITE, eager = true)
+    @Description("apoc.merge.node.eager(['Label'], identProps:{key:value, ...}, config:{onCreateProps:{key:value,...}, onMatchProps:{key:value,...}}) - merge node with dynamic labels, with support for setting properties ON CREATE or ON MATCH")
+    public Stream<NodeResult> nodeEager(@Name("label") List<String> labelNames, @Name("identProps") Map<String, Object> identProps, @Name(value="config", defaultValue = "{}") Map<String, Object> config) {
+        if ((identProps==null) || (identProps.isEmpty())) {
+            throw new IllegalArgumentException("you need to supply at least one identifying property for a merge");
+        }
+
+        validateKeys(config.keySet(), Lists.newArrayList("onCreateProps", "onMatchProps"));
+
+        String labels = labelNames.stream().map(s -> wrapInBacktics(s)).collect(Collectors.joining(":"));
+
+        Map<String, Object> params = new HashMap<>();
+
+        Map<String, Object> onCreateProps = (Map<String, Object>) config.getOrDefault("onCreateProps", Collections.emptyMap());
+        Map<String, Object> onMatchProps = (Map<String, Object>) config.getOrDefault("onMatchProps", Collections.emptyMap());
+
+        params.put("identProps", identProps);
+        params.put("onCreateProps", onCreateProps);
+        params.put("onMatchProps", onMatchProps);
+        String identPropsString = buildIdentPropsString(identProps);
+
+        final String cypher = "MERGE (n:" + labels + "{" + identPropsString + "}) ON CREATE SET n += $onCreateProps ON MATCH SET n += $onMatchProps RETURN n";
+        return db.execute(cypher, params ).columnAs("n").stream().map(node -> new NodeResult((Node) node));
     }
 
     @Procedure(mode = Mode.WRITE)
-    @Description("apoc.merge.relationship(startNode, relType,  {key:value, ...}, {key:value, ...}, endNode) - merge relationship with dynamic type")
+    @Description("apoc.merge.relationship(startNode, relType,  identProps:{key:value, ...}, {key:value, ...}, endNode) - merge relationship with dynamic type")
     public Stream<RelationshipResult> relationship(@Name("startNode") Node startNode, @Name("relationshipType") String relType,
                                                    @Name("identProps") Map<String, Object> identProps, @Name("props") Map<String, Object> props, @Name("endNode") Node endNode) {
         String identPropsString = buildIdentPropsString(identProps);
 
         Map<String, Object> params = new HashMap<>();
```

Comment:
easier with Util.map(args)

Reply:
For building out the `params` map? Got it.